### ISSUE TITLE
Update/extract process fails on 1.8.1

### ISSUE THREAD

**Issue by 0z3l07** [DotBow/Blender-Launcher#41]
**Describe the bug**
Blender Launcher fails to update/extract properly on win 10
Gets stuck on 100% extracting
Running as admin doesnt fix it

**To Reproduce**
Happens on every update process

**Expected behavior**
Update properly/extract

**Screenshots**
![Screenshot 2020-12-13 163622](https://user-images.githubusercontent.com/24977832/102016444-590bce00-3d61-11eb-8f00-f68102be5c58.png)

**System (please complete the following information):**
 - Operation system: win 10
 - Blender Launcher version: 1.8.1
 - Log file: [Blender Launcher.log](https://github.com/DotBow/Blender-Launcher/files/5684880/Blender.Launcher.log)

**Additional context**
None

**Comment by 0z3l07** [comment]
Removing the blender launcher updater.exe and doing the process again fixed it but I get this update error like everytime since you introduced the build in updater.

**Comment by dreamparacite** [comment]
Also Windows 10, can confirm this when I updated to 1.9.0, I tried updating and it stuck at Extracting 100%. After I closed the process and reopened Blender Launcher and tried again the update worked.

**Comment by DotBow** [comment]
Hi! Thanks for report!
@dreamparacite do you have a log with any errors?
I'm not sure if this caused by existing updater file or something else. 
Distributing updater in single app file not an easy task.

**Comment by benthillerkus** [comment]
I've had the same issue but never reported it, as I thought it might be a fluke.
The logs should be next to the executable, right? Because there is a log, but it doesn't have anything in regars to the update process (and the content is older than the update process).

Creating the `.temp` folder works and it does contain the zipped executable.

**Comment by dreamparacite** [comment]
@DotBow 
I didn't have a log after checking today, so I downloaded v1.8.1 and updated but it worked the first time. 
I then downloaded v1.7.1 and updated to v1.9.0 and it also worked the first time.

I started using Blender Launcher around v1.7.0, so it's possible that updating from that to each successive release had a legacy bug I can't reproduce. Will comment with a log if it happens again, though.

**Comment by DotBow** [comment]
> 
> 
> I've had the same issue but never reported it, as I thought it might be a fluke.
> 
> The logs should be next to the executable, right? Because there is a log, but it doesn't have anything in regars to the update process (and the content is older than the update process).
> 
> The updater does create the `.temp` folder and it does contain the zipped executable.

Yeap, right near by BL executable.
It isn't downloading BL into .temp folder. It uses OS user temp folder.
I can't really reproduce the issue. But I know that it exists... 
I think I just should write a new solution because current have unexpected behavior due to heavy file processing.

**Comment by 0z3l07** [comment]
No worries.... Doesn't impact function at all. Either grab the newest release from github or try until it works with the internal updater :grinning:

**Comment by dreamparacite** [comment]
I had an old version of blender launcher on another computer and was able to reproduce the error, here's the log:

```
2020-11-30 15:04:05,442 - Windows - Blender Launcher 1.8.0
Traceback (most recent call last):
  File "windows\main_window.py", line 436, in scraper_finished
RuntimeError: Set changed size during iteration
```

Also I noticed a Windows Defender notification pop up during the update as it suspected a virus, I think it may be flagging the update and interrupting the extract that way.

**Comment by 0z3l07** [comment]
yeah for some reason windows defender now says at of a sudden that the launcher.exe is a trojan ><
Trojan:Win32/Wacatac.B!ml
Worked since 1.0 on my pc....false positive 100%

**Comment by DotBow** [comment]
> 
> 
> I had an old version of blender launcher on another computer and was able to reproduce the error, here's the log:
> 
> ```
> 2020-11-30 15:04:05,442 - Windows - Blender Launcher 1.8.0
> Traceback (most recent call last):
>   File "windows\main_window.py", line 436, in scraper_finished
> RuntimeError: Set changed size during iteration
> ```
> 
> Also I noticed a Windows Defender notification pop up during the update as it suspected a virus, I think it may be flagging the update and interrupting the extract that way.

Nope, this error does nothing with updating. But thanks for help anyway!
Yeap there is some weird Windows Defender behavior, it is false positive, started this evening.

**Comment by DotBow** [comment]
1 out of 5 test updates stuck on extracting, but still no clue why 😞

**Comment by benthillerkus** [comment]
Oh that's interesting - I never had the popup with Windows Defender before. After I got it, I set Blender Launcher to allow, stopped it in the task manager, restarted it, started the update process again and it worked. Am now on 1.9.0

**Comment by 0z3l07** [comment]
So annoying that this is still a thing with 1.9.1->1.9.2 :/

**Comment by DotBow** [comment]
> 
> 
> So annoying that this is still a thing with 1.9.1->1.9.2 :/

Hi! Sorry for the troubles, I will be refactoring auto updater for 1.10.

**Comment by 0z3l07** [comment]
> > So annoying that this is still a thing with 1.9.1->1.9.2 :/
> 
> Hi! Sorry for the troubles, I will be refactoring auto updater for 1.10.

dont worry!

**Comment by DotBow** [comment]
Should be fixed since 1.10.0 release
